pull request for adding vehicle data

BertoGBG opened the pull request:
Closes # (if applicable).

## Changes proposed in this Pull Request
- added Fraunhofer_ISE_vehicles_costs.csv file to inputs folder
- added Appendix-Study-Paths-to-a-Climate-Neutral-Energy-System.pdf file to document folder
- modified snakelike script-->  add: fraunhofer_vehicles_costs
- modified compile_cost_assumptions.py --> 
  1) add: "vehicles" to source_dict
  2) def: "rename_ISE_vehicles" function (similar to "rename_ISE")
  3) add "costs_vehicles" to output files (similar to "costs_ISE")

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Data source for new technologies is cleary stated.
- [ ] Newly introduced dependencies are added to `environment.yaml` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [ ] I consent to the release of this PR's code under the GPLv3 license.

martavp reviewed (changes requested):
Thanks @BertoGBG 

[ ] I think that there is an error in electric vehicles (passengers and trucks are in reverse order). See my comment below. 

[ ] Can you also update the csv files in the output folder?

[ ] The order of magnitude of the cost for the fast charger for electric vehicles is 3 times higher than that of the slow charger. 
I think this should be because, with a fast charger station, you can charge simultaneously many cars at the same time. Could you check if this is the case?
Then, I think we should add the cost of charging 1 car or add the info to the output data on the number of cars that can be charged in a fast charger.

[ ] The O&M cost for trucks is much higher than for passenger cars, I checked and the numbers are correct based on the pdf of the Fraunhofer report, @BertoGBG  do you know why O&M of trucks is higher than for passenger cars?

BertoGBG commented on scripts/compile_cost_assumptions.py:
```diff
@@ -1398,6 +1400,39 @@ def rename_ISE(costs_ISE):
+                          "Lebensdauer": "lifetime",
+                          "M/O-Kosten": "FOM",
+			"Wirkungsgrad*" : "Efficiency (carrier to wheel)",
+			"LKW Batterie-Elektromotor" : "Battery electric (passenger cars)",
+			"PKW Batterie-Elektromotor" : "Battery electric (trucks)",
```
you are correct, it was my mistake.
they should be inverted

BertoGBG commented on scripts/compile_cost_assumptions.py:
```diff
@@ -1398,6 +1400,39 @@ def rename_ISE(costs_ISE):
+			"PKW ICE- Flssigtreibstoff": "Liquid fuels ICE (passenger cars)",
+			"LKW Ladeinfrastruktur Brennstoffzellen Fahrzeuge * LKW": "Charging infrastructure fuel cell vehicles trucks",
+			"PKW Ladeinfrastruktur Brennstoffzellen Fahrzeuge * PKW": "Charging infrastructure fuel cell vehicles passenger cars",
+			"PKW Ladeinfrastruktur schnell  (reine) Batteriefahrzeuge*" : "Charging infrastructure fast (purely) battery electric vehicles passenger cars",
```
I used the official translation from the PDF (in docs). 
I believe that it refers to a fast charging station which does not have a slow charging option (so purely fast charging)

martavp replied:
ok, let's keep the direct translation

BertoGBG commented:
regarding O&M cost for higher for trucks and than passenger cars. 
My guess is based on this
- cost are on % of initial investment 
- same life (y) for both vehicles --> (I disagree btw)
- Trucks drive more distance during life time (600k km vs 150k km is a common the durability target)
- Trucks drive under heavier loads 
- Trucks requirements more maintenance every year

martavp commented:
> regarding O&M cost for higher for trucks and than passenger cars. My guess is based on this
> 
>     * cost are on % of initial investment
> 
>     * same life (y) for both vehicles --> (I disagree btw)
> 
>     * Trucks drive more distance during life time (600k km vs 150k km is a common the durability target)
> 
>     * Trucks drive under heavier loads
> 
>     * Trucks requirements more maintenance every year

Ok, it makes sense to me now.

martavp commented:
> Regarding the order of magnitude of the cost for fast charger compare to slow chargers, I am keen on thinking that the values could be correct, all included.
> 
> 11kW chargers can cost 1000eur and 100kW can be 100 000 eur. For charger only with one outlet, not parallel charging. Some fast charging stations have parallel charging but they divide the current between the vehicles so the current rating is the same. I have seen this figures in the past: This is an example of US charging infrastructure report https://site-assets.evgo.com/f/78437/x/f28386ed92/2020-05-18_evgo-whitepaper_dcfc-cost-and-policy.pdf check Fig.9 .
> 
> However, I am not sure how the cost for fast chargers was estimated by ISE as it is about 6-500 kEur per fast charger. It is possible that they allocated to the single charger also all the costs for building the charging station and extending the grid infrastructure locally (transformers, control systems, digging, project management etc...). But I am speculating.
> 
> I run into similar issues when working in the Volvo group, it was not clear who will pay for those expenses. But in a whole-society model is possible that all those costs were allocated to the chargers.
> 
> I could not find reference in the English version of the study or a method description. I think that those sources are in German, maybe we can ask Sina.

Thanks for the additional info, I agree it is not necessarily an error and it is up the final user to decide how many fast charging stations are assumed by EV.

martavp commented:
Thansks @BertoGBG, IMO this is almost ready to merge, with two additions:

[ ] Can you also update the csv files in the output folder?

[ ] Can you add a small note to the release notes?

martavp commented:
Hi again @BertoGBG ,

I'm trying to use the cost files produced with this PR and I realized there is a bug, 

> costs_vehicles = pd.read_csv(snakemake.input.fraunhofer_vehicles_costs,

should use 
> encoding = "ISO-8859-1")

and the line
> "LKW ICE- Flssigtreibstoff": "Liquid fuels ICE (trucks)",
should be modified.

Otherwise, the ICE cars/trucks are not translated

BertoGBG commented:
Hej @martavp  I have made the change sin the file and tested it locally. 
it works well on my computer.

The translation issue "Flssigtreibstoff" was due to the original csv file in German, in particular the letter ü which was not exported correctly in the csv file.

martavp reviewed (approved):
After the last addition, in my opinion, this is ready to merge.
@BertoGBG can you add a small note to the release notes
@lisazeyen if you can take a look, we can merge this as a first step. And then, we add more updated data on costs.

lisazeyen reviewed (changes requested):
Great thanks @BertoGBG! Just a couple of things which needs to be adjusted before the merge:

1. Can you check that your version is up to date with the current master branch?

There are several technologies missing in the outputs e.g. 

'CH4 (g) pipeline', 'CH4 (g) submarine pipeline', 'H2 (g) pipeline',
       'H2 (g) pipeline repurposed', 'H2 (g) submarine pipeline',
       'electric steam cracker', 'geothermal', 'methanol-to-kerosene',
       'methanol-to-olefins/aromatics'
       Maybe do a comparison of both outputs (current master branch and PR to make sure that there are only the intended changes) 

2. Also I guess you changed on how many digits the values are rounded, this should stay the same as before.

3. Can we change the name of the technology "NT" to something more intuitive? I am not sure what this means.

BertoGBG commented:
Hej
@lisazeyen  I have addressed your comments, 
- the new version is on the updated branch, and the technologies that were missing are now present
- I am not sure what do you mean by the digit values, I did not change anything about those. it is possible that depends on input files have difference.
- 'NT' is removed. It was an error in the input file from Fraunhofer institute. It refers to other technologies (not the vehicles require). but somehow is present in the civ file. I did not modify the csv file because I am not sure if an updated version will have the same issue, but I added a line of code to drop those rows if the index 'NT' is present.

lisazeyen commented on outputs/costs_2020.csv:
```diff
@@ -4,487 +4,519 @@ Ammonia cracker,ammonia-input,1.46,MWh_NH3/MWh_H2,"ENGIE et al (2020): Ammonia t
+Battery electric (trucks),FOM,14.0,%,PATHS TO A CLIMATE-NEUTRAL ENERGY SYSTEM The German energy transformation in its social context. https://www.ise.fraunhofer.de/en/publications/studies/paths-to-a-climate-neutral-energy-system.html,Battery electric (trucks)
+Battery electric (trucks),investment,204067.0,EUR/LKW,PATHS TO A CLIMATE-NEUTRAL ENERGY SYSTEM The German energy transformation in its social context. https://www.ise.fraunhofer.de/en/publications/studies/paths-to-a-climate-neutral-energy-system.html,Battery electric (trucks)
+Battery electric (trucks),lifetime,15.0,years,PATHS TO A CLIMATE-NEUTRAL ENERGY SYSTEM The German energy transformation in its social context. https://www.ise.fraunhofer.de/en/publications/studies/paths-to-a-climate-neutral-energy-system.html,Battery electric (trucks)
+BioSNG,C in fuel,0.32,per unit,"Stoichiometric calculation, doi:10.1016/j.apenergy.2022.120016",
```
If you look at the digits for the values, e.g. here `0.32` compared to what it previously was `0.324`. The values are rounded in the script to a certain value and I assume you changed the rounding to 2 digits instead of 4

martavp replied:
I recreated the output files using this PR and got the 4 digits, so I'm not sure why this is happening. Maybe you open/save the files in excel @BertoGBG ?

BertoGBG replied:
Hej @lisazeyen 
I realised that I didn't upload the new output files before making a pull request. Anyway I am not sure joy the rounding was changed, I never meant to.
 
not the output files are correct with the vehicles added and when I look at them in my GitHub they have 4 digits rounding. 
I suggest you test the code locally if it does not show 4 digits.

/A

lisazeyen commented:
@BertoGBG Great thanks!
